### PR TITLE
Ec 63 update merchant sdk api

### PR DESCRIPTION
**Pull request by motokotomax**
Hide HealthSDK API from client app

### CODE REVIEW

File: MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Core/Adapter/Mapping.swift [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+                                                     lineItems: healthLineItems)
+    }
+}
+
+//MARK: - PaymentProvider
+
+extension PaymentProvider {
+    init(healthPaymentProvider: GiniHealthAPILibrary.PaymentProvider) {
+        let openPlatforms = healthPaymentProvider.openWithSupportedPlatforms.compactMap { PlatformSupported(rawValue: $0.rawValue) }
+        let gpcPlatforms = healthPaymentProvider.gpcSupportedPlatforms.compactMap { PlatformSupported(rawValue: $0.rawValue) }
```
Comment (zladzeyka) [inline MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Core/Adapter/Mapping.swift:56]:
```suggestion
        let gpcSupportedPlatforms = healthPaymentProvider.gpcSupportedPlatforms.compactMap { PlatformSupported(rawValue: $0.rawValue) }
```

File: MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Core/Adapter/Mapping.swift [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        let colors = ProviderColors(healthProviderColors: healthPaymentProvider.colors)
+        let minAppVersions: MinAppVersions?
```
Comment (zladzeyka) [inline MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Core/Adapter/Mapping.swift:58]: [Suggestion]: it's not used anywhere I think we can skip `MinAppVersions`

File: MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Core/GiniMerchant+Test.swift [hunk 3]
```diff
@@ -0,0 +1,17 @@
+//
+//  File.swift
```
Comment (zladzeyka) [inline MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Core/GiniMerchant+Test.swift:2]:
``` suggestion
//  GiniMerchant + Test.swift
```

File: MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Core/GiniMerchant+Test.swift [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+//
+//  Copyright © 2024 Gini GmbH. All rights reserved.
+//
+
+
+@testable import GiniHealthAPILibrary
```
Comment (zladzeyka) [inline MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Core/GiniMerchant+Test.swift:8]: [Clarification]: Shall we move it to tests?

File: MerchantSDK/GiniMerchantSDK/Tests/GiniMerchantSDKTests/MockHelpers/Utils.swift [hunk 5]
```diff
@@ -7,15 +7,30 @@
...
+    var providers: GiniMerchantSDK.PaymentProviders = []
     let providersResponse: [PaymentProviderResponse]? = load(fromFile: fileName)
     guard let providersResponse else { return nil }
     for providerResponse in providersResponse {
         let imageData = UIImage(named: "Gini-Test-Payment-Provider", in: Bundle.module, compatibleWith: nil)?.pngData()
-        let provider = PaymentProvider(id: providerResponse.id, name: providerResponse.name, appSchemeIOS: providerResponse.appSchemeIOS, minAppVersion: providerResponse.minAppVersion, colors: providerResponse.colors, iconData: imageData ?? Data(), appStoreUrlIOS: providerResponse.appStoreUrlIOS, universalLinkIOS: providerResponse.universalLinkIOS, index: providerResponse.index, gpcSupportedPlatforms: providerResponse.gpcSupportedPlatforms, openWithSupportedPlatforms: providerResponse.openWithSupportedPlatforms)
+        let openPlatforms = providerResponse.openWithSupportedPlatforms.compactMap { GiniMerchantSDK.PlatformSupported(rawValue: $0.rawValue) }
+        let gpcPlatforms = providerResponse.gpcSupportedPlatforms.compactMap { GiniMerchantSDK.PlatformSupported(rawValue: $0.rawValue) }
```
Comment (zladzeyka) [inline MerchantSDK/GiniMerchantSDK/Tests/GiniMerchantSDKTests/MockHelpers/Utils.swift:19]:
```suggestion
        let gpcSupportedPlatforms = providerResponse.gpcSupportedPlatforms.compactMap { GiniMerchantSDK.PlatformSupported(rawValue: $0.rawValue) }
```

File: MerchantSDK/GiniMerchantSDK/Tests/GiniMerchantSDKTests/MockHelpers/Utils.swift [hunk 6]
```diff
@@ -7,15 +7,30 @@
...
-    var providers: PaymentProviders = []
+func loadProviders(fileName: String) -> GiniMerchantSDK.PaymentProviders? {
+    var providers: GiniMerchantSDK.PaymentProviders = []
     let providersResponse: [PaymentProviderResponse]? = load(fromFile: fileName)
     guard let providersResponse else { return nil }
     for providerResponse in providersResponse {
         let imageData = UIImage(named: "Gini-Test-Payment-Provider", in: Bundle.module, compatibleWith: nil)?.pngData()
-        let provider = PaymentProvider(id: providerResponse.id, name: providerResponse.name, appSchemeIOS: providerResponse.appSchemeIOS, minAppVersion: providerResponse.minAppVersion, colors: providerResponse.colors, iconData: imageData ?? Data(), appStoreUrlIOS: providerResponse.appStoreUrlIOS, universalLinkIOS: providerResponse.universalLinkIOS, index: providerResponse.index, gpcSupportedPlatforms: providerResponse.gpcSupportedPlatforms, openWithSupportedPlatforms: providerResponse.openWithSupportedPlatforms)
+        let openPlatforms = providerResponse.openWithSupportedPlatforms.compactMap { GiniMerchantSDK.PlatformSupported(rawValue: $0.rawValue) }
```
Comment (zladzeyka) [inline MerchantSDK/GiniMerchantSDK/Tests/GiniMerchantSDKTests/MockHelpers/Utils.swift:18]:
```suggestion
        let openWithPlatforms = providerResponse.openWithSupportedPlatforms.compactMap { GiniMerchantSDK.PlatformSupported(rawValue: $0.rawValue) }
```

**Review by zladzeyka — CHANGES_REQUESTED** [review]
Thank you @motokotomax 👏🏻
Please find a couple of comments 😊
I think it's time to create the GitHub workflows for Merchant 😉

**Comment by motokotomax** [comment]
@zladzeyka  thanks for review!
issues fixed

**Review by zladzeyka — CHANGES_REQUESTED** [review]
@motokotomax  thank you for the fixes 2 more small notes :)

**Review by zladzeyka — APPROVED** [review]
👏🏻

**Pull request merged**